projects/sasil-web/src/logics/hooks/useInfiniteScroll.tsx

```diff
@@ -3,17 +3,17 @@ import { useRef, useEffect, RefObject } from 'react';
 
 type callbackFuncType = (...args: any) => any;
 
-const useInifiniteScroll = (
+const useInfiniteScroll = (
   postsRef: RefObject<HTMLDivElement>,
-  callbackFunc: callbackFuncType,
+  callbackFunc: callbackFuncType | undefined,
 ) => {
   // Node.js 환경에서 빌드됨으로써 생기는 오류 해결하기 위한 참조
   const observerRef = useRef<IntersectionObserver | null>(null);
 
   useEffect(() => {
     observerRef.current = new IntersectionObserver(
       (entries, curObserver) => {
-        if (entries[0].isIntersecting) {
+        if (entries[0].isIntersecting && callbackFunc) {
           callbackFunc(); // 여기서 전역 상태를 업데이트 해야함
 
           curObserver.disconnect(); // 관측 해제
```

Comment:
🍌 혹시 이 분기 처리의 의도가 어떻게 되나요? 이 상황에서는 `callbackFunc`이 있어야만 아래의 `curObserver.disconnect()`도 실행이 되는 것 같은데, 만약 이게 의도가 아니고 단순히 `callbackFunc`을 실행해야 될지 말지를 결정하는 거라면 다음과 같이 고치는 게 나을 것 같아요!

```suggestion
        if (entries[0].isIntersecting) {
          callbackFunc?.(); // 여기서 전역 상태를 업데이트 해야함

          curObserver.disconnect(); // 관측 해제
```

Reply:
작성해준 방법으로 적용해보고 로직 상 문제 없으면 반영할게여!

Reply:
disconnect가 callbackFunc 존재 여부에 상관없이 반영되는게 맞는데, 로직을 잘못 변경했었네여.. 반영 했습니다!